pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1075,10 +1075,23 @@ public CompletableFuture<Optional<Topic>> getTopic(final TopicName topicName, bo
                                             return loadOrCreatePersistentTopic(tpName, createIfMissing,
                                                     properties, topicPolicies);
                                         }
-                                        return CompletableFuture.completedFuture(Optional.empty());
+                                        final String info =
+                                                String.format("Creating a topic encountered an illegal partition name."
+                                                                + " topic_name=%s metadata_partition_number=%s",
+                                                        topicName, metadata.partitions);
+                                        log.warn(info);
+                                        return CompletableFuture
+                                                .failedFuture(new BrokerServiceException.NamingException(info));
                                     });
                         }
                         return loadOrCreatePersistentTopic(tpName, createIfMissing, properties, topicPolicies);
+                    }).thenCompose(optionalTopic -> {
+                        if (!optionalTopic.isPresent() && createIfMissing) {
+                            log.warn("Different topic automatic creation strategies lead to race conditions. "
+                                    + "Try again to try to recover. topic_name={}", topicName);
+                            return getTopic(topicName, true, properties);
```

Comment:
```suggestion
                            return getTopic(topicName, createIfMissing, properties);
```

Reply:
Because we judged the `createIfMissing` condition, therefore, the value should always be `true`

Reply:
Fixed.